### PR TITLE
Mailer

### PR DESCRIPTION
**Pull request by gsyoung2020**
I got the mailer up and running, when in comes to functionality, but wont send without proper settings for a 3rd party

### CODE REVIEW

File: app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  # GET /resource/sign_up
+  def new
+    super
+  end
+
+  # POST /resource
+  def create
+    super
+    # byebug
+    UserMailer.with(user: @user).new_user_email.deliver_now
```
Comment (pjsier) [inline app/controllers/users/registrations_controller.rb:16]: Thanks for digging into this! I think we can keep things a lot more simple by moving this to a callback instead, what do you think? More info here https://stackoverflow.com/a/17480095

File: test/mailers/user_mailer_test.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    assert_emails 1 do
+      email.deliver_now
+    end
+
+    # Check the contents are correct
+    assert_equal ['eattendace@gmail.com'], email.from
+    assert_equal ['eattendace@gmail.com'], email.to
+    assert_equal "You got a new order!", email.subject
+    assert_match user.email, email.html_part.body.encoded
+    assert_match user.email, email.text_part.body.encoded
```
Comment (codeGobbler) [inline test/mailers/user_mailer_test.rb:21]: I'm interested to know how to run these tests, all I've used so far is rspec but I'm sure this stuff is just as important to know if not more. Code looks good so far.

File: spec/helpers/users_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# describe UsersHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe UsersHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (codeGobbler) [inline spec/helpers/users_helper_spec.rb:15]: Just curious, would we still need to write rspec tests for the mailer if it has it's own test(s) @pjsier ?

**Review by codeGobbler — COMMENTED** [review]
Very cool, good job @gsyoung2020 on getting the Mailer up and running. Everything here looks good to me.

**Pull request merged**